Sources/Testing/Events/Recorder/Event.HumanReadableOutputRecorder.swift

```diff
@@ -0,0 +1,378 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Event {
+  /// A type which handles ``Event`` instances and outputs representations of
+  /// them as human-readable messages.
+  ///
+  /// This type can be used compositionally to produce output in other
+  /// human-readable formats such as rich text or HTML.
+  ///
+  /// The format of the output is not meant to be machine-readable and is
+  /// subject to change. For machine-readable output, use ``JUnitXMLRecorder``.
+  public struct HumanReadableOutputRecorder: Sendable {
```

Comment:
I'd like to propose we establish a distinct `@_spi` group for this, such as `@_spi(ExperimentalEventRecording)`. I realize `Event` is already under an SPI group but this is carving out a new subsystem within that domain and it seems appropriate to distinguish them. Also "event _handling_" doesn't feel quite fitting for what this is doing.

Reply:
That's fine with me.